File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/FeedSearchView.java
package com.phicdy.mycuration.presentation.view;

import android.support.annotation.NonNull;

public interface FeedSearchView {
    void startFeedUrlHookActivity(@NonNull String url);
    void showProgressBar();
    void dismissProgressBar();
    void load(@NonNull String url);
    void showInvalidUrlErrorToast();
    void showGenericErrorToast();
    void showAddFeedSuccessToast();
    void finishView();
    void setSearchViewTextFrom(@NonNull String url);
}

File: app/src/main/java/com/phicdy/mycuration/presentation/view/FeedSearchView.kt
package com.phicdy.mycuration.presentation.view

interface FeedSearchView {
    fun startFeedUrlHookActivity(url: String)
    fun showProgressBar()
    fun dismissProgressBar()
    fun load(url: String)
    fun showInvalidUrlErrorToast()
    fun showGenericErrorToast()
    fun showAddFeedSuccessToast()
    fun finishView()
    fun setSearchViewTextFrom(url: String)
    fun trackFailedUrl(url: String)
}
